marlowe-symbolic/src/App.hs

```diff
@@ -63,8 +65,12 @@ handler :: Request -> Context -> IO (Either Response Response)
 handler Request {Req.uuid = u, callbackUrl = cu, contract = c, state = st} context =
   do system "killallz3"
      semaphore <- newEmptyMVar
+     let contract = maybe Close id (JSON.decode (BSU.fromString c))
```

Comment:
seems like a bad idea since if the decoding fails then the analysis will pass, can't we just return a 401 if decoding fails?